### PR TITLE
feat(html): initial html/svg lexer

### PR DESCRIPTION
**Pull request by alexander-akait**
**Description:**

Feature - HTML/SVG lexer/tokenizer.

It is initial version - so there are many rooms to fix/improve:

- no AST/tree-builder currently
- Some TODOs
- many places to improve perf
- it is young so bugs exist
- no `raw`/`quote`/etc fields to better codegen (i.e. output as was written)
- some part of code under `unused` because it will be used in future
- more tests

Based on https://html.spec.whatwg.org/multipage/parsing.html

**BREAKING CHANGE:**

No

**Related issue (if exists):**

No

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,6 +27,7 @@ node_modules/
 
 # Flamegraph
 *.html
+!crates/swc_html*/**/*.html
```
Comment (alexander-akait) [inline .gitignore:30]: Not sure where do we generate htmls files, so using it, ideally we should ignore html files where we generate them

Comment (kdy1) [reply .gitignore:30]: It was used some profiling like `cargo build --timings`

File: crates/swc_html_parser/src/parser/tests.rs [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (kdy1) [inline crates/swc_html_parser/src/parser/tests.rs:1]: I think this files is not required?

File: crates/swc_html_parser/src/tests.rs [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (kdy1) [inline crates/swc_html_parser/src/tests.rs:1]: Same here

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Review by kdy1 — APPROVED** [review]
swc-bump:
 - ''

**Pull request merged**